server/src/main/java/io/crate/execution/engine/aggregation/impl/AverageAggregation.java

```diff
@@ -304,8 +318,7 @@ public Signature boundSignature() {
                         return new AverageState();
                     },
                     (values, state) -> {
-                        state.sum += values.nextValue();
-                        state.count++;
+                        state.addNumber(values.nextValue()); // Mutates state.
```

Comment:
We should probably follow up here and have a version where `sum` is a `long`, to avoid the kahan summation overhead for non-floating numbers.

Reply:
I refactored a little bit to use different state for integral numbers - something similar to HllState but it doesn't store dateType as we already want to reduce fields in HllState and I created a package for average related classes - was a bit hard to scroll over current AverageAggregation as it was already big and with new changes similar to HllAggregation (type aware implementation) it became huge.